numpy/core/src/multiarray/refcount.c

```diff
@@ -274,15 +328,32 @@ _fillobject(char *optr, PyObject *obj, PyArray_Descr *dtype)
             _fillobject(optr + offset, obj, new);
         }
     }
-    else {
-        npy_intp i;
-        npy_intp nsize = dtype->elsize / sizeof(obj);
+    else if (PyDataType_HASSUBARRAY(dtype)) {
+        int size, i, inner_elsize;
+
+        inner_elsize = dtype->subarray->base->elsize;
+        /* Subarrays are always contiguous in memory */
+        size = dtype->elsize / inner_elsize;
 
-        for (i = 0; i < nsize; i++) {
-            Py_XINCREF(obj);
-            NPY_COPY_PYOBJECT_PTR(optr, &obj);
-            optr += sizeof(obj);
+        if (dtype->subarray->base->type_num == NPY_OBJECT) {
+            /* fast path when subarray only includes objects */
```

Comment:
Here I'm a bit less sure, but the slow path would seem fast enough.

Reply:
I timed it in this one and it was quite a big difference. On the other hand, probably almost nobody actually uses these features. And as soon as you do anything else, the actual speed is probably so slow, that it simply does not matter.

So I am good with just removing the fast paths, all of this code is pretty much only used doing creation and removal anyway.

Reply:
+1 from me on removing this fast path.